File: webapp/src/app/models/grades-user-overview.ts
import { Activity, ActivityJson } from 'src/app/models/activity.model';
import { ActivityAverageGrade, ActivityAverageGradeJson } from './activity-average-grade';
import { ActivityItem, ActivityItemLiteJson } from './activity-item.model';
import { FinalGrade, FinalGradeJson } from './final-grade';

export interface GradesUserOverviewJson {
  activities?: ActivityJson[];
  grades?: ActivityItemLiteJson[];
  finalGrade?: FinalGradeJson;
  averageGrades?: ActivityAverageGradeJson[];
}

export interface GradesUserOverview {
  activities?: Activity[];
  grades?: ActivityItem[];
  finalGrade?: FinalGrade;
  averageGrades?: ActivityAverageGrade[];
}

File: api/src/main/java/br/niedunicamp/pojo/GradesUserOverview.java
package br.niedunicamp.pojo;

import java.util.Set;

import br.niedunicamp.model.Activity;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class GradesUserOverview {
    private Set<Activity> activities;
    private Set<ActivityItemLite> grades;
    private FinalGrade finalGrade;
    private Set<ActivityAverageGrade> averageGrades;  
}